### PR TITLE
OV-172: Add azure integration for voices and text to speech

### PR DESCRIPTION
**Pull request by o-nedashkivska**
Related issue - https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/issues/172

Depends on https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/195

### Endpoint Responses Example
**Voices (only `en`):**
```json
{
    "items": [
        {
            "name": "Natasha",
            "shortName": "en-AU-NatashaNeural",
            "gender": "Female",
            "locale": "en-AU",
            "localeName": "English (Australia)",
            "voiceType": "Neural"
        },
        {
            "name": "William",
            "shortName": "en-AU-WilliamNeural",
            "gender": "Male",
            "locale": "en-AU",
            "localeName": "English (Australia)",
            "voiceType": "Neural"
        },
        {
            "name": "Annette",
            "shortName": "en-AU-AnnetteNeural",
            "gender": "Female",
            "locale": "en-AU",
            "localeName": "English (Australia)",
            "voiceType": "Neural"
        }
    ]
}
```
**Text to speech:**
```json
{
    "audioUrl": "cloudfront-audio-url"
}
```

### CODE REVIEW

File: backend/src/common/services/azure-ai/apis/text-to-speech-api.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    public async getVoices(): Promise<Array<AzureGetVoicesResponseDto>> {
+        const response = await this.load(
+            this.getFullEndpoint(TextToSpeechApiPath.VOICES, {}),
+            {
+                method: HTTPMethod.GET,
+                contentType: ContentType.JSON,
+                headers: [
+                    {
+                        key: 'Ocp-Apim-Subscription-Key',
```
Comment (sergiy4) [inline backend/src/common/services/azure-ai/apis/text-to-speech-api.ts:26]: Use constants here, and check my PR for reference.
https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/199

File: backend/src/common/services/azure-ai/azure-ai.service.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        this.subscriptionKey = config.ENV.AZURE.SUBSCRIPTION_KEY;
+        this.serviceRegion = config.ENV.AZURE.SERVICE_REGION;
+    }
+
+    public async getVoices(): Promise<GetVoicesResponseDto> {
+        const response = await this.textToSpeechApi.getVoices();
+
+        const data = response
+            .filter(
+                (data: AzureGetVoicesResponseDto) => data.Locale === 'en-US',
```
Comment (sergiy4) [inline backend/src/common/services/azure-ai/azure-ai.service.ts:40]: Use constant here

File: backend/src/common/services/azure-ai/apis/text-to-speech-api.ts (hunk 1 above)
Comment (o-nedashkivska) [reply backend/src/common/services/azure-ai/apis/text-to-speech-api.ts]: Done

File: backend/src/common/services/azure-ai/azure-ai.service.ts (hunk 2 above)
Comment (o-nedashkivska) [reply backend/src/common/services/azure-ai/azure-ai.service.ts]: Done

**Pull request merged**